src/pages/outlier-detection/OutlierDetection.js

```diff
@@ -262,53 +262,53 @@ class OutlierDetection extends Page {
-    //     const elements = this.state.elements
-    //     for (let i = 0; i < elements.length; i++) {
-    //         const currentElement = elements[i]
-    //         if (currentElement.key === element.key) {
-    //             this.context.updateAppState({
-    //                 pageState: {
-    //                     loading: true,
-    //                 },
-    //             })
-    //             const apiCall = currentElement.marked ? api.delete : api.post
-    //             apiCall(
-    //                 apiConf.endpoints.markDataValue,
-    //                 OutlierAnalyisTable.convertElementToToggleFollowupRequest(
-    //                     currentElement
-    //                 )
-    //             )
-    //                 .then(() => {
-    //                     if (this.isPageMounted()) {
-    //                         currentElement.marked = !currentElement.marked
-    //                         elements[i] = currentElement
-
-    //                         this.context.updateAppState({
-    //                             showSnackbar: true,
-    //                             snackbarConf: {
-    //                                 type: SUCCESS,
-    //                                 message: i18n.t(
-    //                                     currentElement.marked
-    //                                         ? 'Marked'
-    //                                         : 'Unmarked'
-    //                                 ),
-    //                             },
-    //                             pageState: {
-    //                                 elements,
-    //                                 loading: false,
-    //                             },
-    //                         })
-    //                     }
-    //                 })
-    //                 .catch(error => {
-    //                     this.manageError(error)
-    //                 })
-    //             break
-    //         }
-    //     }
-    // }
+    toggleCheckbox(element) {
+        const api = this.context.d2.Api.getApi()
+        const elements = this.state.elements
+        for (let i = 0; i < elements.length; i++) {
+            const currentElement = elements[i]
+            if (currentElement.key === element.key) {
+                this.context.updateAppState({
+                    pageState: {
+                        loading: true,
+                        elements,
+                        showTable: elements && elements.length > 0,
+                    },
+                })
+                const data = OutlierAnalyisTable.convertElementToToggleFollowupRequest(
+                    currentElement
+                )
+                api.update(apiConf.endpoints.markDataValue, data)
+                    .then(() => {
+                        if (this.isPageMounted()) {
+                            currentElement.marked = !currentElement.marked
+                            elements[i] = currentElement
```

Comment:
Technically I think this could be problematic, since `elements` could theoretically have changed while the promise was waiting to resolve - it's a bit safer to duplicate the `elements` array on line 267: `const elements = [...this.state.elements]` OR by grabbing `this.state.elements` here on line 285 (though mutating it might be questionable, idealistically).

Reply:
I've switched to immutable code in e942b5c